fhir-persistence/src/test/java/com/ibm/fhir/persistence/search/test/AbstractSearchIdAndLastUpdatedTest.java

```diff
@@ -0,0 +1,151 @@
+
+        if (DEBUG) {
+            generateOutput(savedResource);
+        }
+
+        List<Resource> resources = runQueryTest(Resource.class, queryParms);
+        assertNotNull(resources);
+        assertEquals(resources.size(), 1, "Number of resources returned");
+        assertTrue(isResourceInResponse(savedResource, resources), "Expected resource not found in the response");
+    }
+
+    @Test
+    public void testSearchWholeSystemUsingIdAndLastUpdatedResource() throws Exception {
+        Map<String, List<String>> queryParms = new HashMap<String, List<String>>();
+        List<String> savedId = Collections.singletonList(savedResource.getId());
+
+        String dateTime = savedResource.getMeta().getLastUpdated().getValue().toString();
+        List<String> savedLastUpdated = Collections.singletonList(dateTime);
+        queryParms.put("_id", savedId);
+        queryParms.put("_lastUpdated", savedLastUpdated);
+
+        if (DEBUG) {
+            generateOutput(savedResource);
+        }
+
+        List<Resource> resources = runQueryTest(Basic.class, queryParms);
+        assertNotNull(resources);
+        assertEquals(resources.size(), 1, "Number of resources returned");
+        assertTrue(isResourceInResponse(savedResource, resources), "Expected resource not found in the response");
+    }
+
+    @Test
+    public void testSearchWholeSystemUsingIdAndLastUpdatedResourceWithSort() throws Exception {
+        Map<String, List<String>> queryParms = new HashMap<String, List<String>>();
+        List<String> savedId = Collections.singletonList(savedResource.getId());
+
+        String dateTime = savedResource.getMeta().getLastUpdated().getValue().toString();
+        List<String> savedLastUpdated = Collections.singletonList(dateTime);
+        queryParms.put("_id", savedId);
+        queryParms.put("_lastUpdated", savedLastUpdated);
+
+        // Sort id and then lastUpdated
+        queryParms.put("_sort", Collections.singletonList("_id,-_lastUpdated"));
+
+        if (DEBUG) {
+            generateOutput(savedResource);
+        }
+
+        List<Resource> resources = runQueryTest(Basic.class, queryParms);
+        assertNotNull(resources);
+        assertEquals(resources.size(), 1, "Number of resources returned");
+        assertTrue(isResourceInResponse(savedResource, resources), "Expected resource not found in the response");
+    }
+
+    /*
+     * generates the output into a resource.
+     */
+    public static void generateOutput(Resource resource) {
+        try (StringWriter writer = new StringWriter();) {
+            FHIRGenerator.generator(Format.JSON, true).generate(resource, System.out);
+            System.out.println(writer.toString());
+        } catch (FHIRGeneratorException e) {
+            fail("unable to generate the fhir resource to JSON", e);
+        } catch (IOException e1) {
+            fail("unable to generate the fhir resource to JSON (io problem) ", e1);
+        }
+    }
+}
```

Comment:
Could you add one "gt" test for _lastModified?

Reply:
sure

Reply:
in next commit